fhirstore/search/corequerybuilder.py

```diff
@@ -0,0 +1,136 @@
+import sys
+import re
+import json
+
+from collections import defaultdict
+from collections.abc import Mapping
+
+number_prefix_matching = {"gt": "gt", "ge": "gte", "lt": "lt", "le": "lte"}
+
+
+def build_element_query(key, value):
+    """Translate and parse search parameters (key, value) to an
+    elasticSearch element query
+    """
+    element_query = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+
+    if isinstance(value, int) or isinstance(value, float):
+        element_query["match"][key] = value
+
+    else:
+        numeric_prefix = re.search(r"^(gt|lt|ge|le)([0-9].*)$", f"{value}")
+        eq_prefix = re.search(r"^(eq)([0-9].*)$", f"{value}")
+        special_prefix = re.search(r"^(ne|sa|eb|ap)([0-9].*)$", f"{value}")
+        pipe_suffix = re.search(r"(.*)\|(.*)", value)
+
+        string_modif = re.search(
+            r"^(.*):(contains|exact|above|below|not|in|not-in|of-type|identifier)$", key
+        )
+        if key == "_text" or key == "_content":
+            element_query["query_string"]["query"] = value
+        elif string_modif:
+            string_modifier = string_modif.group(2)
+            string_field = string_modif.group(1)
+            if string_modifier == "contains":
+                element_query["query_string"]["query"] = f"*{value}*"
+                element_query["query_string"]["default_field"] = string_field
+
+            elif string_modifier == "exact":
+                element_query["term"] = {string_field: value}
+
+            elif string_modifier == "not":
+                element_query["bool"]["must_not"]["match"][string_field] = f"{value}"
+
+            elif string_modifier == "not-in":
+                element_query["simple_query_string"]["query"] = f"-{value}"
+                element_query["simple_query_string"]["fields"] = [string_field]
+
+            elif string_modifier == "in":
+                element_query["query_string"]["query"] = f"{value}"
+
+            elif string_modifier == "below":
+                element_query["simple_query_string"]["query"] = f"({value})*"
```

Comment:
Do you know the difference between simple_query_string and query_string?
If we could always use one, maybe we could simplify the string_modif case. But we can do that later if you prefer and finish this PR first 🙂